### PR TITLE
SEQNG-600 Resizable columns on the queue table

### PR DESCRIPTION
**Pull request by cquiroz**
This PR updates the queue table to make the columns resizable. 

The code to handle the columns has been simplified and made more general

Here's a video showing the result
https://monosnap.com/file/68ZtJzBpSxerzBO422bRL8JqVPNpf4#

### CODE REVIEW

**Comment by andrewwstephens** [comment]
Awesome!

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/resources/less/style.less [hunk 1]
```diff
@@ -645,6 +645,42 @@ body {
   overflow: unset !important;
   pointer-events: auto;
 }
+
+//===========================
+// Styles for column dragging
+.DragHandle {
+  flex: 0 0 16px;
+  z-index: 2;
```
Comment (sraaphorst) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/resources/less/style.less]: I'm guessing `z-index` is the depth of something, but what is the `flex`?

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala [hunk 2]
```diff
@@ -0,0 +1,533 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
```
Comment (sraaphorst) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: You should probably change this in your editor. 😉

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/resources/less/style.less (hunk 1 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/resources/less/style.less]: flex is a layout system in css 
https://www.w3schools.com/css/css3_flexbox.asp

We use it extensively so it's worth learning it. this is a good course:

https://scrimba.com/g/gflexbox

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala (hunk 2 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: We should change on the build I think

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+          <.span(^.cls := "DragHandleIcon", "⋮")
+        )
+    )
+
+  def rowClassName(p: Props)(i: Int): String =
+    ((i, p.rowGetter(i)) match {
+      case (-1, _) =>
+        SeqexecStyles.headerRowStyle
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s == SequenceState.Completed  =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowPositive
```
Comment (sraaphorst) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: I assume this means `SeqexecStyles` has a monoid type class? Nice.

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+      case (_, QueueRow(_, s, _, _, _, _, _)) if s.isError                     =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowNegative
+      case (_, QueueRow(_, s, _, _, _, active, _)) if active && !s.isInProcess =>
+        SeqexecStyles.stepRow |+| SeqexecStyles.rowActive
+      case _                                                                   =>
+        SeqexecStyles.stepRow
+    }).htmlClass
+
+  // scalastyle:off
+  private def columns($ : $, size: Size): List[Table.ColumnArg] = {
```
Comment (sraaphorst) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: I think I may have asked you this before a long time ago, but what does:

`$ : $`

mean?

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala (hunk 3 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: `GStyle`

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala (hunk 4 above)
Comment (cquiroz) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: This is for historical convention though it is a bad name. `$` basically gives you access to the react `context`, including props and stat

Comment (cquiroz) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: I could rename it to `backend` if it is clearer

Comment (sraaphorst) [reply modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/QueueTable.scala]: It's fine 👍 . I'll just take note of it.

**Pull request merged**